src/wallet/rpcwallet.cpp

```diff
@@ -550,7 +550,11 @@ static UniValue signmessage(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_TYPE_ERROR, "Address does not refer to key");
     }
 
-    const SigningProvider* provider = pwallet->GetSigningProvider();
+    CScript script_pub_key = GetScriptForDestination(*pkhash);
```

Comment:
I don't think it really matters as we still need `pkhash` since `signmessage` is supposed to only work on p2pkh addresses.

Reply:
> we still need `pkhash` since `signmessage` is supposed to only work on `p2pkh` addresses

That will change in https://github.com/bitcoin/bitcoin/pull/16440/files#diff-df7d84ff2f53fcb2a0dc15a3a51e55ceR559